airflow/kubernetes/pod_generator_deprecated.py

```diff
@@ -0,0 +1,652 @@
+# Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
Do we really need a separate file here? We just want to deprecate but still support  `KubernetesExecutor` config. The code you delete in `airflow/kubernetes/pod_generator.py` can just go in a separate function or a static method (or just in that if block) in the same file, just prefix it with `_`

Reply:
@kaxil We need a separate file because in the simplify_kubernetes_executor PR we change a lot more stuff in the pod_generator file and it is safest to just keep it separated. This PR is mostly about simplifying that PR.

Reply:
aah should I review that first?

Reply:
Or we should separate the "new module" bit in a different PR because most-likely more reviewers would face the same question